### PR TITLE
`Development`: Fix an issue in the statistics service when a course has no exercises

### PR DESCRIPTION
**Pull request by dfuchss**
#### General

~-  I tested **all** changes and their related features with **all** corresponding user types on a test server.~
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).

### Motivation and Context

Optional.get() with courses that have no exercises leads to an internal server error & minor code improvements

See https://github.com/ls1intum/Artemis/pull/4926/files#diff-759a2ec63cde05b2a2476a1aae1df8b7d49fc4c07da1624887e8643dae556d68L132

### Description
see motivation

### Steps for Testing

Prerequisites:
- 1 Admin

1. Create a new course with no exercises
2. Open the course statistics
3. No internal server error should be shown

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by akesfeden — DISMISSED** [review]
Tested on TS1, I didn't see any Internal Server Error notification and the Network tab of DevTools didn't contain any failed request to the application server.

File: src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java [hunk 1]
```diff
@@ -122,14 +122,22 @@ public List<Integer> getChartData(SpanType span, Integer periodIndex, GraphType
...
     public CourseManagementStatisticsDTO getCourseStatistics(Long courseId) {
 
         var courseManagementStatisticsDTO = new CourseManagementStatisticsDTO();
         Set<Exercise> exercises = statisticsRepository.findExercisesByCourseId(courseId);
-        Course course = exercises.stream().findAny().get().getCourseViaExerciseGroupOrCourseMember();
+
+        if (exercises.isEmpty()) {
+            // Handle newly created courses that have no exercises
+            courseManagementStatisticsDTO.setAverageScoreOfCourse(0.0);
+            courseManagementStatisticsDTO.setAverageScoresOfExercises(new ArrayList<>());
```
[the remark this answers is not in the document]
Comment (dfuchss) [reply src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java:136]: I was not sure about this. some lines below it was also `new ArrayList<>()`

[the remark this answers is not in the document]
Comment (dfuchss) [reply src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java:136]: https://github.com/ls1intum/Artemis/pull/4926/files#diff-759a2ec63cde05b2a2476a1aae1df8b7d49fc4c07da1624887e8643dae556d68R164 @Strohgelaender

[the remark this answers is not in the document]
Comment (dfuchss) [reply src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java:136]: Collections.emptyList() is immutable as far as I remember.

[the remark this answers is not in the document]
Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java:136]: Yes, it is immutable. But we directly return this object and should not modfiy this list later, or do I miss something here?

[the remark this answers is not in the document]
Comment (dfuchss) [reply src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java:136]: I don't know whether some constraints exist here. I simply saw that the method seems to return a modifiable list.
I can change this behavior but was not sure whether something might break :)

[the remark this answers is not in the document]
Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java:136]: I just checked it again and you are right, the other used List is in fact modifyable. But this `getCourseStatistics()`-method is only used in one place (`StatisticsResource`) where it is directly mapped into an `ResponseEntity`. So there are arguments for both approaches, and if you want to keep it modifyable consistently I'm fine with it.

[the remark this answers is not in the document]
Comment (dfuchss) [reply src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java:136]: If we can use an immutable lists I'm fine with it. I'll push it in a minute :)

[the remark this answers is not in the document]
Comment (dfuchss) [reply src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java]: Done

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**